packages/server/src/arpa_reporter/services/validate-upload.js

```diff
@@ -157,19 +157,46 @@ function validateIdentifier(recipient, recipientExists) {
     const hasUEI = Boolean(recipient.Unique_Entity_Identifier__c);
     const hasTIN = Boolean(recipient.EIN__c);
     const entityType = recipient.Entity_Type_2__c;
-    const isContractorOrBeneficiary = (entityType.includes('Contractor') || entityType.includes('Beneficiary'));
+    const isContractor = entityType.includes('Contractor');
+    const isBeneficiary = entityType.includes('Beneficiary');
     const isSubrecipient = entityType.includes('Subrecipient');
+    const isIAA = entityType.includes('IAA');
 
     if (isSubrecipient && !recipientExists && !hasUEI) {
         errors.push(new ValidationError(
             'UEI is required for all new subrecipients',
             { col: 'C', severity: 'err' },
         ));
-    } else if (isContractorOrBeneficiary && !hasUEI && !hasTIN) {
+    } else if (isContractor && !hasUEI && !hasTIN) {
         errors.push(new ValidationError(
             'At least one of UEI or TIN/EIN must be set, but both are missing',
             { col: 'C, D', severity: 'err' },
         ));
+    } else if (isBeneficiary && recipientExists) {
+        if (recipientExists.created_at < new Date('2024-07-01') && !hasTIN && !hasUEI) {
```

Comment:
Now that we are using `recipientExists` as an object, it may make sense to rename it to not seem like a boolean.